xarray/core/variable.py

```diff
@@ -1166,16 +1175,39 @@ def to_index(self):
         if isinstance(index, pd.MultiIndex):
             # set default names for multi-index unnamed levels so that
             # we can safely rename dimension / coordinate later
-            valid_level_names = [name or '{}_level_{}'.format(self.name, i)
+            valid_level_names = [name or '{}_level_{}'.format(self.dims[0], i)
                                  for i, name in enumerate(index.names)]
             index = index.set_names(valid_level_names)
         else:
             index = index.set_names(self.name)
         return index
 
     @property
+    def level_names(self):
+        """Return MultiIndex level names or None if this IndexVariable has no
+        MultiIndex.
+        """
+        index = self.to_index()
+        if isinstance(index, pd.MultiIndex):
+            return index.names
+        else:
+            return None
+
+    @level_names.setter
+    def level_names(self, value):
```

Comment:
You don't really need this -- if you define a property without a setter, you get `AttributeError` when you try to set it, anyways